lib/core/shared/providers.dart

```diff
@@ -19,6 +25,17 @@ final imagePickingRemoteServiceProvider =
           ref.watch(imagePickerProvider),
           ref.watch(firebaseStorageProvider),
         ));
-final imagePickingRepositoryProvider = Provider((ref) =>
-    ImagePickingRepository(ref.watch(imagePickingRemoteServiceProvider),));
+final imagePickingRepositoryProvider = Provider((ref) => ImagePickingRepository(
+      ref.watch(imagePickingRemoteServiceProvider),
+    ));
+
+// Product providers
+final productListRemoteServiceProvider =
+    Provider((ref) => ProductListRemoteService(ref.watch(firestoreProvider)));
+
+final productListRepoProvider = Provider((ref) =>
+    ProductListRepository(ref.watch(productListRemoteServiceProvider)));
 
+final productListNotifierProvider =
+    StateNotifierProvider<ProductListNotifier, ProductListState>(
+        (ref) => ProductListNotifier(ref.watch(productListRepoProvider)));
```

Comment:
The product list shouldn't be in core, coz retailer use stream of product, but consumer use future